src/main/java/org/broadinstitute/hellbender/utils/SATagAlignment.java

```diff
@@ -0,0 +1,122 @@
+    String mapQ;
+    String NM;
+
+    public String getContig() {
+        return contig;
+    }
+
+    public void setContig(final String contig) {
+        this.contig = contig;
+    }
+
+    public String getPos() {
+        return pos;
+    }
+
+    public void setPos(final String pos) {
+        this.pos = pos;
+    }
+
+    public String getStrand() {
+        return strand;
+    }
+
+    public void setStrand(final String strand) {
+        this.strand = strand;
+    }
+
+    public String getCigar() {
+        return cigar;
+    }
+
+    public void setCigar(final String cigar) {
+        this.cigar = cigar;
+    }
+
+    public String getMapQ() {
+        return mapQ;
+    }
+
+    public void setMapQ(final String mapQ) {
+        this.mapQ = mapQ;
+    }
+
+    public String getNM() {
+        return NM;
+    }
+
+    public void setNM(final String NM) {
+        this.NM = NM;
+    }
+
+    // builder for SARead that parses the existing SA tag
+    private SATagAlignment(String SATag) {
+        String[] values = SATag.split(",", -1);
+        if (values.length != 6) {
+            throw new GATKException("Could not parse SATag: "+SATag);
+        }
+        if (!values[1].equals("*") && Integer.parseInt(values[1]) < 0) {
+            throw new GATKException("Could not parse POS in SATag: "+SATag);
+        }
+        if (!values[3].matches(SATagAlignment.cigarRe)){
+            throw new GATKException("Could not parse cigar in SATag: " + SATag);
+        }
+        if (!values[4].equals("*") && Integer.parseInt(values[4]) < 0) {
+            throw new GATKException("Could not parse MapQ in SATag: " + SATag);
+        }
+        this.contig = values[0];
+        this.pos = values[1];
+        this.strand = values[2];
+        this.cigar = values[3];
+        this.mapQ = values[4];
+        this.NM = values[5];
+    }
+
+    public SATagAlignment(GATKRead read) {
+        Utils.nonNull(read);
+
+        this.contig = read.getContig();
+        this.pos = read.getStart()+"";
+        this.strand = (read.isReverseStrand()?"-":"+");
+        this.cigar = read.getCigar().toString();
+        this.mapQ = read.getMappingQuality()+"";
+        this.NM = (read.hasAttribute("NM")? read.getAttributeAsString("NM"):"*");
+    }
+
+    public String toString() {
+        return String.format("%s,%s,%s,%s,%s,%s;",
+                ((this.contig!=null)?this.contig:"*"),
+                ((this.pos!=null)?this.pos:"0"),
+                (strand.equals("-")?"-":"+"),
+                ((this.cigar!=null)?this.cigar:"*"),
+                ((this.mapQ!=null)?this.mapQ:"255"),
+                ((this.NM!=null)?this.NM:"*"));
+    }
+
+    final private static String cigarRe = "\\*|([0-9]+[MIDNSHPX=])+";
+
+    public static List<SATagAlignment> parse(String saTag) {
+        String[] tags = saTag.split(";");
+        final List<SATagAlignment> supplementaryReads = Arrays.stream(tags).map(r -> new SATagAlignment(r)).collect(Collectors.toList());
```

Comment:
you don't need a named variable here.  just return Arrays.stream(tags).... directly.